Jenkinsfile

```diff
@@ -1,29 +1,61 @@
-node {
-    def commit_id;
-    def RUST = '$HOME/.cargo/bin/';
-    def RUST_TESTING_FLAGS ='RUSTFLAGS="-Zprofile -Ccodegen-units=1 -Cinline-threshold=0 -Clink-dead-code -Coverflow-checks=off -Zno-landing-pads" CARGO_INCREMENTAL=0';
-    stage('Preparation') {
-        checkout scm
-        sh "git rev-parse --short HEAD > .git/commit-id"
-        commit_id = readFile('.git/commit-id').trim()
-        sh "${RUST}cargo --version"
+pipeline {
+    agent { 
+         docker { image 'tul1/avalon-rs:latest' }
     }
-    stage('Build') {
-        sh "${RUST}cargo build"
-    }
-    stage('Unit tests') {
-        sh "${RUST_TESTING_FLAGS} ${RUST}cargo test --lib"
-    }
-    stage('Integration tests') {
-        sh "${RUST_TESTING_FLAGS} ${RUST}cargo test --test integration_tests"
-    }
-    if(env.BRANCH_NAME == 'master') {
-        stage('Publish crate') {
-            sh "echo publish"
-            // sh "cargo publish"
+    stages {
+        stage('Rustfmt') {
+            steps {
+                // This step isn't currently available because
+                // cargo fmt is not passing due to some technical
+                // dept we've gathered.
+                // TODO uncomment once cargo fmt passes. 
+                // sh "cargo fmt -- --check"
+                sh "echo coming soon"
+            }
+        }
+        stage('Clippy') {
+            steps {
+                // TODO uncomment once cargo clippy passes. 
+                // sh "cargo clippy"
+                sh "echo coming soon"
+            }
+        }
+        stage('Build') {
+            steps {
+                sh "cargo build"
+            }
+        }
+        stage('Test') {
+            steps {
+                // TODO uncomment once cargo test passes. 
+                // sh "cargo test"
+                sh "echo coming soon"
```

Comment:
Are we merging this if the test are not running? Can't we make this a draft PR until that is solved so we don't loose our existing CI?

Reply:
Tests were solved inside this pr https://github.com/tul1/avalon-rs/pull/28 . we can wait until it's merge. You cannot make a pr a draft pr.

Reply:
OK. Let's wait until that one is merged

Reply:
uncommneted